enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1008 @@
+               - subnet-0fcf8e0392f0910d1
+   [...]
+   ---
+   apiVersion: operator.openshift.io/v1
+   kind: IngressController
+   metadata:
+     name: default
+     namespace: openshift-ingress-operator
+   spec:
+     [...]
+     endpointPublishingStrategy:
+       type: LoadBalancerService
+       loadBalancer:
+         scope: External
+         providerParameters:
+           type: AWS
+           aws:
+             type: Classic
+             classicLoadBalancer:
+               subnets:
+                 ids:
+                 - subnet-0fcf8e0392f0910d0
+                 - subnet-0fcf8e0392f0910d1
+     [...]
+   ```
+3. The `default` IngressController is configured with the specified subnets, and user-created
+   IngressControllers created on Day 2 will default to using the subnets specified in the Ingress Config.
+
+#### Using AWS Subnet Discovery for IngressControllers with BYO VPC during Installation Workflow
+
+A cluster admin wants to install a cluster into an existing VPC, but still have the AWS CCM
+automatically discover the subnets for its IngressController's load balancer: 
+
+1. Cluster admin creates an install-config with `platform.aws.subnetsConfig` specified
+   with the subnet(s) while ensuring `platform.aws.subnetsConfig[].roles` is configured
+   with only `Cluster`:
+    ```yaml
+    apiVersion: v1
+    baseDomain: devcluster.openshift.com
+    metadata:
+      name: my-cluster
+    [...]
+    platform:
+      aws:
+        region: us-east-2
+        subnetsConfig:
+        - id: subnet-0fcf8e0392f0910d2
+          roles:
+          - Cluster
+        - id: subnet-0fcf8e0392f0910d3
+          roles:
+          - Cluster
+    [...]
+    ```
+2. The OpenShift Installer installs the cluster into the VPC where the cluster-role subnets exist, but does not
+   configure any specific subnets for the `default` IngressController or `cluster` Ingress Config manifests.
+3. When the `default` IngressController is created, the load balancer will automatically map to the appropriately
+   tagged subnets in the VPC (see [AWS Subnet Discovery](#aws-subnet-discovery) for how subnets are discovered).
+
+#### Specifying Dedicated IngressController-role Subnets During Installation with a BYO VPC Workflow
+
+Specifying dedicated IngressController subnets enables cluster admins to isolate
+IngressController subnets from the node subnets (i.e. cluster-role subnets):
+
+1. Cluster admin creates an install-config with `platform.aws.subnetsConfig`, specifying
+   each subnet to have either the `IngressController` or `Cluster` role in `platform.aws.subnetsConfig.roles`, but
+   not both:
+    ```yaml
+    apiVersion: v1
+    baseDomain: devcluster.openshift.com
+    metadata:
+      name: my-cluster
+    [...]
+    platform:
+      aws:
+        region: us-east-2
+        subnetsConfig:
+        - id: subnet-0fcf8e0392f0910d4 # AZ us-east-1
+          roles:
+          - Cluster
+        - id: subnet-0fcf8e0392f0910d5 # AZ us-east-1
```

Comment:
@gcs278 - is the scope of this feature supporting only those subnets in the same AZ or targeting subnets created in different ones too? Example `IngressController`-role subnet created in `us-east-1a`, and `Cluster`-role subnet created in `us-east-1b`, placing only those two in the install-config as the example in this snippet.

Suggestion as subnets are created in AZ:

```suggestion
        - id: subnet-0fcf8e0392f0910d4 # AZ us-east-1a
          roles:
          - Cluster
        - id: subnet-0fcf8e0392f0910d5 # AZ us-east-1a
```

Reply:
This feature supports any combination of AZs for IngressController & Cluster roles (as long it's in the same VPC of course). So yes, you can have a IngressController-role in a different AZ than a Cluster-role subnet. This enables the "dedicated ingress controller subnet" use case (per Service Delivery's request).

Added suggestion.

Reply:
Okay, so I would recommend to exercise the scenarios where subnets added to the ingress LB are different than nodes, and document AWS expected health check behavior somewhere here and/or in the product document if needed. Something like:

```yaml
    platform:
      aws:
        region: us-east-2
        subnetsConfig:
        - id: subnetID-private-use1a # AZ us-east-1a
          roles:
          - Cluster
        - id: subnetID-private-use1b # AZ us-east-1b
          roles:
          - IngressController
        lbType: NLB
```

IIRC we've discussed on Slack that scenario, and LBs will mark target as `unused`. In general that state would not be expected for operational/health targets. I have some questions: is the health check works as expected? how to remediate that (do we need)? do we have impact in the routing traffic? and in the operation(users seeing that unexpected health check state in AWS Console), do we need to provide additional information explaining about that behavior?

I know some of answers we've discussed there, but would be nice to have documented here too.

Reply:
Added a installer validation section `Reject IngressControllers AZs that do not match Cluster AZs`, take look and let me know if it makes sense. I think it answers your questions by just rejecting these scenarios.